latch/registry/table.py

```diff
@@ -464,19 +549,27 @@ def commit(self) -> None:
         """
         updates: List[l.SelectionNode] = []
 
-        self._add_record_upserts_selection(updates)
+        coalesced = self._coalesce_mutations()
+
+        for muts in coalesced:
+            if all(isinstance(mut, _TableRecordsUpsertData) for mut in muts):
+                self._add_record_upserts_selection(muts, updates)
+            if all(isinstance(mut, _TableRecordsDeleteData) for mut in muts):
+                self._add_record_deletes_selection(muts, updates)
```

Comment:
I see why you are using coalesce_mutations -- however I would just inline it. The code with the all(...) stuff is very weird even if it works in this case.